system/modules/isotope/library/Isotope/Frontend.php

```diff
@@ -630,36 +630,39 @@ public static function getPagesInCurrentRoot(array $arrPages, $objMember = null)
      */
     public function addProductToBreadcrumb($arrItems)
     {
-        if (Input::getAutoItem('product', false, true) != '') {
-            $objProduct = Product::findAvailableByIdOrAlias(Input::getAutoItem('product', false, true));
-
-            if (null !== $objProduct) {
-
-                /** @var \PageModel $objPage */
-                global $objPage;
-                global $objIsotopeListPage;
-
-                $last = count($arrItems) - 1;
-
-                // If we have a reader page, rename the last item (the reader) to the product title
-                if (null !== $objIsotopeListPage) {
-                    $arrItems[$last]['title'] = $this->prepareMetaDescription($objProduct->meta_title ? : $objProduct->name);
-                    $arrItems[$last]['link']  = $objProduct->name;
-                } // Otherwise we add a new item for the product at the last position
-                else {
-                    $arrItems[$last]['href'] = \Controller::generateFrontendUrl($arrItems[$last]['data']);
-                    $arrItems[$last]['isActive'] = false;
-
-                    $arrItems[] = array(
-                        'isRoot'   => false,
-                        'isActive' => true,
-                        'href'     => $objProduct->generateUrl($objPage),
-                        'title'    => $this->prepareMetaDescription($objProduct->meta_title ? : $objProduct->name),
-                        'link'     => $objProduct->name,
-                        'data'     => $objPage->row(),
-                    );
-                }
-            }
+        if (!($alias = Input::getAutoItem('product', false, true))
+            || ($objProduct = Product::findAvailableByIdOrAlias($alias)) === null
+        ) {
+            return $arrItems;
+        }
+
+        /** @var \PageModel $objPage */
+        global $objPage;
+
+        if ($objPage->type === 'error_404') {
```

Comment:
we should do this check before the *expensive* search for product in the database.